sqlx-postgres/src/arguments.rs

```diff
@@ -68,10 +69,12 @@ impl PgArguments {
             .push(value.produces().unwrap_or_else(T::type_info));
 
         // encode the value into our buffer
-        self.buffer.encode(value);
+        self.buffer.encode(value)?;
```

Comment:
Same thing here as with `MySqlArguments::add()`.

Reply:
Resetting the buffer after encoding fails is now done for all 3 databases!